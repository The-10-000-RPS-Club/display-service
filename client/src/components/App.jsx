import React from 'react';
import axios from 'axios';
import styled from 'styled-components';

import Pictures from './pictures';
import Description from './description';
import Name from './name';
import Star from './rating';
import Price from './price';
import Color from './color';
import Size from './size';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      product: {
        id: 1,
        product_name: 'Tasty Fresh Car',
        rating: 4,
        ratingsAmt: 421,
        price: '186.48',
        color: 'maroon',
        description: 'The automobile layout consists of a front-engine design, with transaxle-type transmissions mounted at the rear of the engine and four wheel drive',
        image: 'https://fecclothes.s3-us-west-2.amazonaws.com/Project/1.jpg',
        clothing_type: 'pants',
      },
    };
    this.getData = this.getData.bind(this);
  }

  componentDidMount() {
    this.getData(1);
  }

  getData(id) {
    axios.get(`/api/products/${id}`)
      .then(({ data }) => {
        this.setState({
          product: data,
        });
      });
  }

  render() {
    const { product } = this.state;
    return (
      <div>
        <Wrapper>
          <Name name={product.product_name} />
          <Star ratings={product.rating} id={product.id} ratingsAmt={product.ratingsAmt} />
          <Price price={product.price} />
          <Color color={product.color} price={product.price} />
          <Size size={product.clothing_type} />
        </Wrapper>
        <Image>
          <Pictures image={product.image} />
        </Image>
        <Description description={product.description} />
      </div>
    );
  }
}
const Image = styled.div`
    margin: 150px 0px 0px 600px;
`;

const Wrapper = styled.div`
    float: right;
    margin: 5px 400px 0px 50px;
    height: 700px;
    width: 400px;
    `;
export default App;
