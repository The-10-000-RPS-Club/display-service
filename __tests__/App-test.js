/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount, render } from 'enzyme';

import App from '../client/src/components/App';

describe('A suite', function() {
  it('should render without throwing an error', function() {
    expect(shallow(<App />).contains('Hello from React')).toBe(true);
  });

  xit('should be selectable by class "foo"', function() {
    expect(shallow(<App />).is('.foo')).toBe(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<App />).find('.foo').length).toBe(0);
  });

  it('should render to static HTML', function() {
    expect(render(<App />).text()).toEqual('Hello from React');
  });
});