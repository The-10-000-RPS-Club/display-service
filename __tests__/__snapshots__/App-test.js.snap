// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite should render without throwing an error 1`] = `
<div
  className="a5mel5-0 iRpgjs"
>
  <p>
    The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design
  </p>
</div>
`;

exports[`A suite should render without throwing an error 2`] = `
<div>
  <div
    className="sdphye-0 ghDmRY"
  >
    <div>
      REI Co-op
    </div>
  </div>
  <div
    className="sdphye-1 jAjLYf"
  >
    <span>
       REI Co-op  
    </span>
    <span>
      Ergonomic Rubber Hat
    </span>
  </div>
</div>
`;

exports[`A suite should render without throwing an error 3`] = `
<img
  className="nuhrh2-0 bpmCqm"
/>
`;

exports[`A suite should render without throwing an error 4`] = `
<div>
  <div
    className="sc-179h418-1 iTihlm"
  />
  <span>
    4 | 292
  </span>
  <span
    className="sc-179h418-0 hPXnqg"
  >
    Item  #1
  </span>
</div>
`;

exports[`A suite should render without throwing an error 5`] = `
<div
  className="sc-5tcji-0 gNWCqj"
>
  <span>
    $186.48
  </span>
  <div
    className="sc-5tcji-1 gZZJKx"
  >
    <p>
      Orders of $50 or more ship for free.
    </p>
  </div>
</div>
`;

exports[`A suite should render without throwing an error 6`] = `
<div
  className="sgaphz-0 kbROOT"
>
  <div>
    Color : maroon
  </div>
  <div
    className="sgaphz-1 ckHUWv"
  >
    <div>
      $186.48
    </div>
  </div>
  <div
    className="sgaphz-2 fvaWoO"
  />
</div>
`;
